modules/atom-ide-ui/pkg/atom-ide-diagnostics-ui/lib/DiagnosticsViewModel.js

```diff
@@ -107,15 +112,20 @@ export class DiagnosticsViewModel {
         isVisible,
         diagnostics: this._filterDiagnostics(
           globalState.diagnostics,
+          globalState.descriptions,
           instanceState.textFilter.pattern,
           instanceState.hiddenGroups,
           globalState.filterByActiveTextEditor
             ? globalState.pathToActiveTextEditor
             : null,
         ),
+        descriptions: globalState.descriptions,
         onTypeFilterChange: this._handleTypeFilterChange,
         onTextFilterChange: this._handleTextFilterChange,
-        selectMessage: this._selectMessage,
+        selectMessage: (message: DiagnosticMessage) => {
+          this._selectMessage(message);
+          globalState.onSelectMessage(message);
+        },
```

Comment:
hm, it's a bummer we're making this function every time we see an update, even if `globalState.onSelectMessage` hasn't changed. i guess we could do something like `selectMessage: createSelectMessage(globalState.onSelectMessage),` where

```js
const createSelectMessage = memoizeUntilChanged(
  onSelectMessage => message => {
    this._model.setState({selectedMessage: message});
    onSelectMessage(message);
  }
);
```

(then we would just get rid of `_selectMessage()`.) what do you think?